lib/session.go

```diff
@@ -31,9 +31,36 @@ func (e *Session) Assume(arn string) (*Session, error) {
 	}
 
 	duration := expiration.Sub(time.Now())
-	creds, err := e.AWSCreds.AssumeRole(arn, duration)
+
+	var creds *AWSCredentials
+
+	arnStruct, err := ParseARN(arn)
+
 	if err != nil {
-		return nil, err
+		identityArn, err := e.AWSCreds.GetCallerIdentity()
+		if err != nil {
+			return nil, err
+		}
+
+		roleArn := &ARN{
+			Partition: identityArn.Partition,
+			Service:   "iam",
+			Region:    identityArn.Region,
+			AccountId: identityArn.AccountId,
+			Resource:  "role/" + arn,
+		}
+
+		roleArnString := roleArn.String()
+		fmt.Println("assumed ARN is :", roleArnString)
+		creds, err = e.AWSCreds.AssumeRole(roleArnString, duration)
```

Comment:
If the calculated role is successfully assumed here, it needs to be passed down to the `&Session{}` instantiation below (line 68). It's used later by the `Session` object to inject metadata into the child process.